test/test_jit.py

```diff
@@ -5001,6 +5001,87 @@ def test(op, const, swap_args):
     def test_tensor_number_math(self):
         self._test_tensor_number_math()
 
+    def test_torch_tensor_bad_input(self):
+        with self.assertRaisesRegex(RuntimeError, "Input list to torch.tensor must be of ints, floats, "
+                                    "or bools, got None"):
+            @torch.jit.script
+            def test():
+                return torch.tensor([None])
+
+        with self.assertRaisesRegex(RuntimeError, "Note: empty lists are constructed as Tensor\[\]"):
+            @torch.jit.script
+            def foo():
+                return torch.tensor([])
+
+        @torch.jit.script
+        def foo():
+            return torch.tensor([[2, 2], [1]])
+        with self.assertRaisesRegex(RuntimeError, "Expected sequence of length"):
+            foo()
+
+    def test_torch_tensor_empty_list(self):
+        code = dedent('''
+        def func():
+            return torch.tensor(torch.jit.annotate(List[int], []))
+        ''')
+        scope = {}
+        exec(code, globals(), scope)
```

Comment:
Does this have to use `exec`? It looks like it would be fine as a normal @torch.jit.script function

Reply:
You don't need to exec this, the `exec` is useful when one needs to format a string to produce a test case but that doesn't occur here